### PR TITLE
*: use a Helm chart instead of kustomize

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Having to expect what the namespace and name *would* end up being for kustomize after deployment to refer to them in the VAP and VAPB, etc, was not ideal. Using Helm instead is simple.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Fixes #39

File: config/helm/templates/leader_election_role.yaml [hunk 1]
```diff
@@ -1,15 +1,11 @@
-# permissions to do leader election.
 apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
 metadata:
   labels:
-    app.kubernetes.io/name: role
-    app.kubernetes.io/instance: leader-election-role
```
Comment (stevekuznetsov) [inline config/helm/templates/leader_election_role.yaml:7]: Note on the labels - `instance` is supposed to identify the specific installation instance, if you had a `wordpress` app that were installed ten times on the cluster, `instance` is the name of each installation. Not useful for us.

File: config/helm/templates/leader_election_role.yaml [hunk 2]
```diff
@@ -1,15 +1,11 @@
...
-    app.kubernetes.io/component: rbac
```
Comment (stevekuznetsov) [inline config/helm/templates/leader_election_role.yaml:8]: Note on the labels - `component` is supposed to be the semantic name of the component within the app, so for a web-app, you'd have a databse, a server, etc. Not useful for us and certainly not meant to identify that an RBAC object is rbac ;)

File: config/helm/templates/leader_election_role.yaml [hunk 3]
```diff
@@ -1,15 +1,11 @@
...
-    app.kubernetes.io/created-by: msi-acrpull
```
Comment (stevekuznetsov) [inline config/helm/templates/leader_election_role.yaml:9]: `created-by` and `part-of` are redundant really with the name of the app, no huge need to add these

File: config/helm/templates/controller_role.yaml [hunk 4]
```diff
@@ -1,8 +1,7 @@
----
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
-  name: manager-role
+  name: acrpull-controller
```
Comment (stevekuznetsov) [inline config/helm/templates/controller_role.yaml:4]: With `v1beta2` this controller knows how to provide pull credentials for WI as well as MI, so `msi-acrpull` was no longer the best name.

File: config/helm/templates/validatingadmissionpolicyparameters.yaml [hunk 5]
```diff
@@ -7,12 +7,11 @@ apiVersion: v1
...
   labels:
-    app.kubernetes.io/component: validatingadmissionpolicy
-    app.kubernetes.io/created-by: msi-acrpull
-    app.kubernetes.io/part-of: msi-acrpull
-    app.kubernetes.io/managed-by: kustomize
+    app.kubernetes.io/name: acrpull
+    app.kubernetes.io/managed-by: Helm
 data:
-  controllerName: 'msi-acrpull-controller-manager'
-  controllerNamespace: 'msi-acrpull-system'
```
Comment (stevekuznetsov) [inline config/helm/templates/validatingadmissionpolicyparameters.yaml:17]: Great example of what was annoying with `kustomize` - you had to know beforehand that `kustomize` would prefix `msi-acrpull-` to the namespace and name of things, then record that here while the actual SA name & namespace were something different. Not great UX.

File: test/Makefile [hunk 6]
```diff
@@ -91,35 +91,49 @@ _output/config:
...
 _output/deploy/image: _output/image
 _output/deploy/previous-image: _output/previous-image
-_output/deploy/image _output/deploy/previous-image: _output/aks.kubeconfig $(KUSTOMIZE) _output/deploy/prometheus-crds _output/system-vmss-puller.json _output/deploy/crd
-	cd _output/config/manager && $(KUSTOMIZE) edit set image controller="$(shell cat _output/$(notdir $@) )" && cd -
-	$(KUSTOMIZE) build _output/config/default | kubectl --kubeconfig _output/aks.kubeconfig apply --server-side -f -
-	kubectl --kubeconfig _output/aks.kubeconfig --namespace msi-acrpull-system wait --for condition=Available deployment/msi-acrpull-controller-manager --timeout 120s
+_output/deploy/image _output/deploy/previous-image: _output/aks.kubeconfig $(HELM) _output/deploy/prometheus-crds _output/system-vmss-puller.json
+	cp values.template.yaml _output/values.yaml
+	sed -i -e 's|IMAGE_DIGEST|$(shell cat _output/$(notdir $@) )|' _output/values.yaml
+	$(HELM) template --kubeconfig _output/aks.kubeconfig acrpull ./../config/helm --values _output/values.yaml --create-namespace --dry-run=server > _output/chart.yaml
```
Comment (stevekuznetsov) [inline test/Makefile:102]: Spent a long time fighting with Helm over who controlled which release and when, and it was honestly not that useful or interesting. `helm template | kubectl apply` is all we needed.

File: test/values.template.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+image: IMAGE_DIGEST
+affinity:
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
```
Comment (stevekuznetsov) [inline test/values.template.yaml:4]: It was nice to get this test-specific code broken out of the templates.

File: config/helm/templates/controller_role.yaml (hunk 4 above)
Comment (weinong) [reply config/helm/templates/controller_role.yaml:4]: WI is still MI. It's just a different credential flow :P

**Review by weinong — APPROVED** [review]
LGTM

**Pull request merged**